### ISSUE TITLE
Is there a way to convert an image to base64 first as part of pre-request script?

### ISSUE THREAD

**Issue by Sjain01** [a85/POSTMan-Chrome-Extension#6370]
Hi Reader,

I want to test batch of images via the rest calls. The only condition is those images have to be base64 encoded. So I am wondering if it is possible to first convert an image to base64 as part of pre-request script and then send the encoded string in request body.

I searched about it a lot and found CryptoJS function but using that I couldn't encode an image.
I placed the paths of all image files in a data file and tried using those paths in CryptoJS function as follows :- 

`console.log("encrypt: " + encrypt()); // 'NzUzMjI1NDE='

function encrypt() {
  // PROCESS
  const encryptedWord = CryptoJS.enc.Utf8.parse(pm.iterationData.get("blurryImage")); // encryptedWord Array object
  const encrypted = CryptoJS.enc.Base64.stringify(encryptedWord); // string: 'NzUzMjI1NDE='
  return encrypted;
}`

where blurryImage is a variable in data file. but instead of picking up the path , it is using the literal value of variable name.

Please advise how can this be achieved.

**Comment by shamasis** [comment]
I’m afraid that this is not possible using scripts in Postman. Postman scripts can not access files from inside sandbox. Maybe you could ask this question at https://community.getpostman.com where someone might have found alternative ways to achieve this.

**Comment by shamasis** [comment]
As a note for future reference of this feature request, we could open up now a shim of fs using working directory restricted variant from inside sandbox.

**Comment by joanbarros** [comment]
I tried creating a Visualizer to display the base64 images but when I passed the base64 content to a img tag I got this error:

```
Refused to load the image 'data:image/png;base64,iVBO....QmCC' because it violates the following Content Security Policy directive: "img-src http: https:".
```

Any way of telling Postman to allow this?

PD: @Sjain01 not sure if this would also resolve your particular need. But thought I should share. Maybe it's useful for you if the permission issues can be resolved somehow.

**Comment by Zhigalin** [comment]
Yeah, it would be nice to being able to access filesystem from the pre-request script

**Comment by Lampei** [comment]
Running into this blocker myself now.  Having the working directory available would allow the CI/CD build process to move necessary files around that can be referenced from within the test scripts, and even perhaps allow requires without having to package and install them onto the server.

**Comment by dcoffey3296** [comment]
+1 on this feature, thanks

**Comment by prajwalrajup** [comment]
+1

**Comment by rgfretes** [comment]
+1

**Comment by iplaksiy** [comment]
+1 

Much needed feature in both the pre-request and tests script.

**Comment by LexGrishin** [comment]
+1

**Comment by K-RutviK-K** [comment]
@postman-support  add this feature in future update

**Comment by mariiakhairi** [comment]
+1

**Comment by KYKong97** [comment]
+1

**Comment by cdddg** [comment]
+1

**Comment by sanghh111** [comment]
I think build a simple api for convert Image to base64 Image.

**Comment by agravity-philipp** [comment]
+1 (at least save files to a pre-defined folder would be enough)
My use case would be to store it and use it as an input to multipart form.
Would be happy if postman could realize this!
